lib.go

```diff
@@ -129,6 +131,28 @@ func Not(a BoolPredicate) BoolPredicate {
 
 // GetFieldByTag returns a field from the object based on the tag.
 func GetFieldByTag(ival interface{}, tagName string, fieldNames []string) (interface{}, error) {
+	i, err := getFieldByTag(ival, tagName, fieldNames)
+	if err == nil {
+		return i, nil
+	}
+
+	var nfe *notFoundError
+	if errors.As(err, &nfe) {
+		return nil, trace.NotFound(nfe.Error())
+	}
```

Comment:
suggestion: fieldNames are available here in the top level call, we could build the message here and in `getFieldByTag` just use a single error value that doesn't store anything specific, and compare with `errors.Is`

Reply:
That did cross my mind, I was thinking the error message would actually be more helpful if it included all the field names, but yeah there may be some back compat considerations. Either way is still a great improvement, no need to micro-optimize errors.Is vs errors.As with the amount of reflection this function is still doing